docs/reference/api.yml

```diff
@@ -209,20 +216,95 @@ paths:
         all packaging_shapes containing "fe" will be returned. 
         This is useful if you have a search in your application, 
         for a specific product field.
+  /api/v3/product:
+    parameters: []
+    post:
+      summary: Create or update product
+      operationId: post-api-v3-product-barcode
+      responses:
+        '200':
+          description: OK
+      description: |-
+        New proposed API to send structured product data in a JSON format instead of in a flattened list of key / value pairs field as-in the current product add / edit API that relies on a multipart/form-data format.
+
+        This new API will be used in particular to send structured packaging data: https://github.com/openfoodfacts/openfoodfacts-server/blob/main/docs/explanations/packaging-data.md
+
+        The new API can then be gradually extended to support other product fields.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                userid:
+                  type: string
+                password:
+                  type: string
+                code:
+                  type: string
+                product:
+                  $ref: ./schemas/product_update_api_v3.yaml
```

Comment:
Don't we need the language code ? Or do we consider it to be embeded in the url (but I don't think this is practical from a client point of view).

Reply:
Good point, I'll add it.